### PR TITLE
pre-commit: Remove reduntant checks.

### PR DESCRIPTION
**Pull request by daavoo**
Already covered by `ruff`.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,23 +45,6 @@ repos:
...
-          - flake8-debugger
-          - flake8-string-format
-          - flake8-typing-imports
-  - repo: https://github.com/PyCQA/bandit
-    rev: '1.7.5'
-    hooks:
-    - id: bandit
-      args: ["-c", "pyproject.toml", "--quiet", "--format=custom"]
-      files: ^dvc/
-      additional_dependencies: [".[toml]"]
```
Comment (skshetry) [inline .pre-commit-config.yaml:70]: There are some shell/subprocess specific checks that are absent in ruff.

https://github.com/charliermarsh/ruff/issues/1646

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -51,23 +45,6 @@ repos:
...
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-bugbear
-          - flake8-comprehensions
-          - flake8-debugger
-          - flake8-string-format
-          - flake8-typing-imports
```
[the remark this answers is not in the document]
Comment (daavoo) [reply .pre-commit-config.yaml:63]: Do we care much about it to block this? When would that happen and not be caught in CI?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (daavoo) [reply .pre-commit-config.yaml:70]: Do you think those are critical? Is it worth keeping it or would they get implemented soon?

Comment (daavoo) [reply .pre-commit-config.yaml:70]: Seems that we are explicitly ignoring in the code the occurrences of some of those checks

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (skshetry) [reply .pre-commit-config.yaml:63]: That's the whole point of having pre-commit checks.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (skshetry) [reply .pre-commit-config.yaml:70]: Let's wait for a new ruff release. Looks like some checks were added.

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (skshetry) [reply .pre-commit-config.yaml:63]: Also cc'ing @pmrowla regarding flake8 checks.

**Comment by skshetry** [comment]
@daavoo, please update to the latest version of ruff, and let's remove bandit/flake8 config and merge this. If you don't mind, let's get rid of isort as well, preferably in a separate PR.

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmrowla) [reply .pre-commit-config.yaml:63]: I'm fine with dropping flake8

**Comment by skshetry** [comment]
Closing for now, please reopen when we are in code-freeze for 3.0.

**Pull request closed without merging**